### ISSUE TITLE
Skill template should reference same NuGet package versions used by the Virtual Assistant

### ISSUE THREAD

**Issue by dipanjan77** [microsoft/botframework-solutions#858]
**Project**

Virtual Assistant
**Description**

Adding a new skill with Skill Template 4.0.5.37 breaks your project build. The Nuget references on your new skill project are of ver 4.2.0 but other projects in your solution references 4.2.2.
The compiler complains of a downgrade. If you update all versions of your dependencies from NuGet Manager, you end up with more versioning issues. 

**To Reproduce**

- Install the Skill Template 4.0.5.37
- Add a new Skill Project
- Reference the Bot solution project to your Skill project and its test and cross reference it to Virtual Assist project.
- Build the solution and it fails

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by ctstone** [comment]
As a workaround, you can open the NuGet Package Manager Console and run

```PowerShell
Update-Package -ProjectName {SkillProjectName}
```

In case the shared libraries (Microsoft.Bot.*) versions have advanced beyond the version in your solution (currently 4.2.2) you may have to get more creative with the `update-package` command to target specific versions or update the packages across the entire solution.

**Comment by dipanjan77** [comment]
Thanks @ctstone !!
Yes that had been tried and it solves the version problem, but the custom skill still fails to build:
`Error	CS1739	The best overload for 'TelemetryLoggerMiddleware' does not have a parameter named 'logUserName'`

On debugging I find that the signatures of the method TelemetryLoggerMiddleware does not match as implemented in VirtualAssistant project. It seems the master commit where this change was made has either regressed or it may have not been pushed yet. 
If I change this code as follows, I can build the custom skill, but not sure further down the road what implication it will have:

```
//var appInsightsLogger = new TelemetryLoggerMiddleware(telemetryClient, logUserName: true, logOriginalMessage: true);
var appInsightsLogger = new TelemetryLoggerMiddleware(telemetryClient, logPersonalInformation: true);
```

**Comment by ctstone** [comment]
I think that is the correct fix. If we compare the same file in CalendarSkill, we see the same:
```csharp
var appInsightsLogger = new TelemetryLoggerMiddleware(telemetryClient, logPersonalInformation: true);
```

Also, I think `Startup.cs` will only ever execute if the skill is run as a top level web server (opposed to loading the assembly dynamically from the VirtualAssistant project.

So there are really two bugs with the template:
1. Skill template should reference same NuGet package versions used by the Virtual Assistant (4.2.2 instead of 4.2.0)
1. Skill template uses incorrect `TelemetryLoggerMiddleware` parameters in `Startup.cs`

**Comment by lauren-mills** [comment]
Closed by #879
